### PR TITLE
Docs: further clarify heroku config settings

### PR DESCRIPTION
**Pull request by toranb**
I did my first Heroku deploy this week and had a heck of a time with the buildpack configuration for Phoenix 1.14.10 and wanted to be sure others who follow get up and running with ease :)

I've got a 100% functional Elixir app on heroku [now](https://elixirprimes.herokuapp.com/) w/ the prod [config](https://github.com/toranb/elixir-primes-demo/blob/master/config/prod.exs#L12-L17) shown in this PR so I know it works today with the very latest Phoenix. Related [discussion](https://github.com/HashNuke/heroku-buildpack-elixir/issues/161) on the buildpack github for anyone interested.

### CODE REVIEW

**Comment by toranb** [comment]
Can someone re-kick the travis build at some point? Strange when a doc PR fails the build 😆

**Comment by petermm** [comment]
Think you missed the Procfile part in the guide https://hexdocs.pm/phoenix/heroku.html#adding-the-phoenix-static-buildpack

"The Procfile provided by the Elixir Buildpack runs the command mix run --no-halt. The Phoenix Buildpack changes it to the proper mix phx.server, which we have to manually configure when not using the Phoenix Buildpack. To do so, create a file called Procfile in the root directory and add a command to run Phoenix server with this line:"

**Comment by toranb** [comment]
@petermm in the README for the buildpack under features they list ```If your app doesn't have a Procfile, default web task mix run --no-halt will be run.``` so I assumed .... no Procfile was required to get up and running

File: guides/deployment/heroku.md [hunk 1]
```diff
@@ -175,6 +175,16 @@ url: [scheme: "https", host: "mysterious-meadow-6277.herokuapp.com", port: 443],
 force_ssl: [rewrite_on: [:x_forwarded_proto]],
 ```
 
+Also, regardless if you are using distillery or mix releases, the buildpack requires that you set `server: true` so the prod.exs configuration should appear something like this.
+
+```elixir
+config :hello, Hello.Endpoint,
+  server: true,
```
Comment (josevalim) [inline guides/deployment/heroku.md]: `server: true` should not be necessary on Heroku as long as the server is being started with `mix phx.server` (as `mix phx.server` does it for you). So can you please double check if Heroku is starting your app with `mix phx.server`?

Comment (toranb) [reply guides/deployment/heroku.md]: @josevalim It seems to be using `mix run --no-halt` from what I can see in the logs on Heroku. 

```
2019-11-15T18:34:52.174322+00:00 heroku[web.1]: Starting process with command `mix run --no-halt`
2019-11-15T18:34:56.399341+00:00 app[web.1]: 18:34:56.399 [info] Running GameWeb.Endpoint with cowboy 2.7.0 at :::55169 (http)
2019-11-15T18:34:56.409063+00:00 app[web.1]: 18:34:56.408 [info] Access GameWeb.Endpoint at https://elixirmatch.herokuapp.com
```

Here is the single [commit](https://github.com/toranb/workshop-deployed/commit/5b33d1e1f89d9f522a8fbafe6c6b2606ff83eba2) w/ my buildpack and mix config changes

Comment (josevalim) [reply guides/deployment/heroku.md]: Then you are running into the issue described here: https://github.com/phoenixframework/phoenix/pull/3597/files#diff-58026df9e0782f005bac538b6685333eR121

You either have only one buildpack or they are in the wrong order. Using the correct order should fix it. :)

**Comment by josevalim** [comment]
We will clarify the docs to make sure you need both buildpacks, thank you for the input.

**Pull request closed without merging**